pkg/executables/cmk.go

```diff
@@ -0,0 +1,363 @@
+package executables
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"gopkg.in/ini.v1"
+
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/templater"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+const (
+	cmkConfigFileName             = "cmk_tmp.ini"
+	cloudStackb64EncodedSecretKey = "CLOUDSTACK_B64ENCODED_SECRET"
+	cloudmonkeyInsecureKey        = "CLOUDMONKEY_INSECURE"
+)
+
+var (
+	//go:embed config/cmk.ini
+	cmkConfigTemplate string
+	requiredEnvsCmk   = []string{cloudStackb64EncodedSecretKey, cloudmonkeyInsecureKey}
+)
+
+const (
+	maxRetriesCmk    = 5
+	backOffPeriodCmk = 5 * time.Second
+)
+
+type Cmk struct {
+	writer     filewriter.FileWriter
+	executable Executable
+	retrier    *retrier.Retrier
+	execConfig *cmkExecConfig
+}
+
+func (c *Cmk) SearchTemplate(ctx context.Context, domain string, zone string, account string, template string) (string, error) {
+	templates, err := c.ListTemplates(ctx, template)
+	if err != nil {
+		return "", fmt.Errorf("search template %s error: %v", template, err)
+	} else if len(templates) > 1 {
+		return "", fmt.Errorf("duplicate templates %s found", template)
+	} else if len(templates) == 0 {
+		return "", fmt.Errorf("template %s not found", template)
+	}
+	return templates[0].Name, nil
```

Comment:
What is the difference between the input `template` and this? Is it like full path or something like that?

Reply:
input template may be either a template name or ID. This will always return the human readable name